services/core-api/app/api/mines/mine/models/mine_type.py

```diff
@@ -45,3 +53,91 @@ def expire_record(self):
             detail.expire_record()
         self.active_ind = False
         self.save()
+
+    @classmethod
+    def update_mine_type_details(cls,
+                                 mine_type_guid=None,
+                                 mine_guid=None,
+                                 permit_guid=None,
+                                 mine_tenure_type_code=None,
+                                 mine_disturbance_codes=[],
+                                 mine_commodity_codes=[]):
+        mine_type = None
+
+        def find_mine_type(mine_type_guid, mine_guid, permit_guid):
+            if mine_type_guid:
+                return cls.find_by_guid(mine_type_guid)
+            elif mine_guid:
+                return cls.query.filter_by(mine_guid=mine_guid).first()
+            elif permit_guid:
+                return cls.query.filter_by(permit_guid=permit_guid).first()
+            else:
+                raise Exception('mine_guid or mine_type_guid or permit_guid must be specified.')
+
+        mine_type = find_mine_type(mine_type_guid, mine_guid, permit_guid)
+
+        if not mine_type:
+            raise Exception('Site Property does not exist')
+        
+        if mine_tenure_type_code: 
+            mine_type.mine_tenure_type_code = mine_tenure_type_code
+            
+        existing_disturbance_codes = [
+            dist_code.mine_disturbance_code for dist_code in mine_type.mine_type_detail
```

Comment:
`detail.mine_disturbance_code for detail in mine_type.mine_type_detail` may be more accurate

Reply:
done